packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-dependencies.js

```diff
@@ -148,12 +148,19 @@ async function asyncWhere(array, filter) {
     return (await Promise.all(array.map(async el => ((await filter(el)) ? [el] : [])))).flat();
 }
 
-function isPeerDependencyOptional(name, map = {}) {
+function isPeerDependencyOptional(name, packageJson = {}) {
     // certain peerDependencies can be optionally available, described in package.json `peerDependencyMeta`.
     // https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta
+    const map = packageJson.peerDependenciesMeta || {};
     if (map[name]) {
         return !!map[name].optional;
     }
 
     return false;
 }
+
+function isDependencyOptional(name, packageJson = {}) {
+    // certain dependencies can be optionally available, described in package.json `optionalDependencies`.
+    // https://docs.npmjs.com/cli/v7/configuring-npm/package-json#optionaldependencies
+    return Object.prototype.hasOwnProperty.call(packageJson.optionalDependencies || {}, name);
```

Comment:
Can we use this suggestion instead?
```
return name in (packageJson.optionalDependencies || {});
```

Reply:
Howdy! This was my first idea, although the benefit of the current code is that, compared to the `in` operator, it will only check against the object's own properties, versus the propeties of the `prototype`. 

Lets say that a NPM dependency is called `toString`, if we use the `in` operator, this function will return true, which isn't what we want.

What are your thoughts?

Reply:
The chances for that to happen and to be combined with a native dependency is very very low. But if you think is safer, for me is fine. I would check also with @tamelbrun but he is off tomorrow..

Reply:
Using just `(packageJson.optionalDependencies || {}).hasOwnProperty(name)` also works in the same way..

<img width="926" alt="image" src="https://user-images.githubusercontent.com/45102481/132564513-c9b58dab-0d05-4b8a-95fb-4c83ff9e628d.png">

Reply:
The odds are small, yes, yet with the current implementation there's no risk. I do think it's safer. 

Using `hasOwnProperty` produces an eslint error (from eslint:recommened): `no-prototype-builtins`: https://eslint.org/docs/rules/no-prototype-builtins

Reply:
The JSON object does have the regular prototype in practice, but as the ESLint rule points out, I guess we don't really know that. (I had to look at the source of the underlying library to know for sure that it was a regular `JSON.parse()`.) So at first glance, the `Object.prototype.hasOwnProperty.call` construct seems like the only solution that is strictly correct.

I have a radically different suggestion though: on line 98, many properties of `packageJson` are already extracted as an array of keys using `Object.keys()`, which avoids the protoype problem altogether. If we do the same for `packageJson.optionalDependencies`, we can simply do `optionaDependencies.includes(name)` on line 105. You could even rewrite the `peerDependenciesMeta` check to `const peerDependenciesOptional = Object.entries(packageJson.peerDependenciesMeta).filter(e => !!e[1].optional).map(e => e[0]);`, then you can do a simple `includes()` for that one as well.

Let's see if I can figure out how this "suggesting a change" feature on GitHub works :P.

Reply:
Morning @tamelbrun. IIUC, adding `optionalDependencies` to the list of `dependencies` (line 98) could change behaviour; we currently do not process (bundle) `optionalDependencies`, yet in some cases, a dependency is listed in `dependencies` *and* `optionalDependencies`, and the current implementation takes this into account whrereby it doesn't throw when this is the case (rollup cannot find an optional dependency).

I've taken the rest of your suggestion into account and adjusted the implementation; it reduces code (yay, thanks!). Please let me know what you think.

Reply:
My suggestion was only about using `Object.keys()`,  not about changing the existing list of dependencies. But now that you mentiion it: if you like code reduction, it would make sense to change those into `Object.keys(... || {})` as well ;).

Reply:
Ah, nice. Changed!

Reply:
✨